dissect/target/plugins/os/unix/bsd/osx/user.py

```diff
@@ -0,0 +1,48 @@
+import plistlib
+from typing import Iterator
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+
+AccountPolicyRecord = TargetRecordDescriptor(
+    "osx/account_policy",
+    [
+        ("string", "user"),
+        ("string", "generateduid"),
+        ("datetime", "creation_time"),
+        ("datetime", "password_last_time"),
+        ("datetime", "failed_login_time"),
+        ("varint", "failed_login_count"),
+        ("path", "source"),
+    ],
+)
+
+
+class UserPlugin(Plugin):
+    # TODO: Parse additional user data like: HeimdalSRPKey, KerberosKeys, ShadowHashData, LinkedIdentity,
+    # inputSources, smartCardSecureTokenData, smartCardSecureTokenUUID, unlockOptions, smb_sid
+
+    USER_PATH = "/var/db/dslocal/nodes/Default/users"
+    USER_GLOB = "/var/db/dslocal/nodes/Default/users/*.plist"
```

Comment:
Wouldn't it be nicer to use the now functional `self.target.user_details.all_with_home()` for this?

Reply:
I have chosen for `self.target.user_details.all()`, because the data is not retrieved from the home folder and it could be possible that an user is does not have a home directory. 


https://developer.apple.com/documentation/foundation/filemanager/1642853-homedirectory/
https://apple.stackexchange.com/questions/96665/is-it-possible-to-create-a-user-without-a-home-directory

Reply:
I will add a test and adjust the function because of this.